src/components/main.vue

```diff
@@ -80,20 +80,24 @@
           </template>
         </Recommendations>
       </div>
-
       <ResultsList v-if="$x.totalResults" v-infinite-scroll:body-scroll>
-        <template #layout="{ results }">
-          <BaseVariableColumnGrid
-            #default="{ item: result }"
-            class="x-grid"
-            :animation="resultsAnimation"
-            :items="results"
-          >
-            <Result :result="result" />
-          </BaseVariableColumnGrid>
-        </template>
-      </ResultsList>
+        <BannersList>
+          <PromotedLists>
+            <BaseVariableColumnGrid class="x-grid" :animation="resultsAnimation">
+              <template #Result="{ item: result }">
+                <Result :result="result" />
+              </template>
+              <template #Banner="{ item: banner }">
+                <Banner :banner="banner" />
+              </template>
 
+              <template #Promoted="{ item: promoted }">
+                <Promoted :promoted="promoted" />
+              </template>
+            </BaseVariableColumnGrid>
+          </PromotedLists>
```

Comment:
Change to `PromotedsList`

```suggestion
          <PromotedsList>
            <BaseVariableColumnGrid class="x-grid" :animation="resultsAnimation">
              <template #Result="{ item: result }">
                <Result :result="result" />
              </template>
              <template #Banner="{ item: banner }">
                <Banner :banner="banner" />
              </template>
              <template #Promoted="{ item: promoted }">
                <Promoted :promoted="promoted" />
              </template>
            </BaseVariableColumnGrid>
          </PromotedsList>
```